go/types/map_iterator.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package types
+
+import (
+	"github.com/attic-labs/noms/go/d"
+)
+
+// MapIterator can efficiently iterate through a Noms Map.
+type MapIterator struct {
+	cursor    *sequenceCursor
+	LastKey   Value
+	LastValue Value
```

Comment:
Can we remove LastKey/LastValue until we need them? They were for implementing the fancy SetIterator compositions, but we're not composing MapIterators.